src/Rings/mpoly.jl

```diff
@@ -383,7 +383,11 @@ function singular_generators(B::IdealGens, monorder::MonomialOrdering=default_or
   isdefined(B, :ord) && B.ord == monorder && monomial_ordering(B.Ox, Singular.ordering(base_ring(B.S))) == B.ord && return B.gens.S
   SR = singular_poly_ring(B.Ox, monorder)
   f = Singular.AlgebraHomomorphism(B.Sx, SR, gens(SR))
-  return Singular.map_ideal(f, B.gens.S)
+  S = Singular.map_ideal(f, B.gens.S)
+  if isdefined(B, :ord) && B.ord == monorder
+    S.isGB = B.isGB
+  end
+  return S
```

Comment:
This should fix the main problem in #4018: `singular_generators` would not carry the information that the polynomials form a Gröbner basis with respect to `monorder` over to the Singular side.